src/exampleWorker.js

```diff
@@ -0,0 +1,229 @@
+/* eslint-env browser */
+
+'use strict';
+
+self.importScripts('../dist/ohm.js', 'utils.js');
+
+// Web Worker that generates examples.
+// communicates with 'exampleGenerationUI.js'
+
+(function(root, initModule) {
+  if (typeof exports === 'object') {
+    module.exports = initModule;
+  } else {
+    initModule(root.ohm, root.utils);
+  }
+})(this, function(ohm, utils) {
+  postMessage('WORKER STARTED');
+  var grammar;
+
+  var examplePieces = {};
+  var generator;
+
+  var overrides = {
+    ident: function(g, ex, syn, args) {
+      return {needHelp: true};
+    },
+    ctor: function(g, ex, syn, args) {
+      return {needHelp: true};
+    },
+    letter: function(g, ex, syn, args) {
+      if (Math.floor(Math.random() * 2) === 0) {
+        return {example: String.fromCharCode(Math.floor(26 * Math.random()) + 'a'.charCodeAt(0))};
+      } else {
+        return {example: String.fromCharCode(Math.floor(26 * Math.random()) + 'A'.charCodeAt(0))};
+      }
+    }
+  };
+
+  self.addEventListener('message', function(e) {
+    postMessage(e.data.name);
+    switch (e.data.name) {
+      case 'initialize':
+        /* eslint-disable no-eval */
+        grammar = ohm.makeRecipe(eval(e.data.recipe));
+        /* eslint-enable no-eval */
+
+        start();
+        break;
+      case 'request:examples':
+        var ruleName = e.data.args[0];
+
+        var examplesForRule = examplePieces[ruleName] || null;
+        self.postMessage({name: 'received:examples',
+                          args: [ruleName, examplesForRule]});
+        break;
+      case 'update:neededExamples':
+        var neededExamples = utils.difference(
+          Object.keys(grammar.rules),
+          Object.keys(examplePieces)
+        );
+
+        self.postMessage({name: 'received:neededExamples',
+                          args: [neededExamples]});
+        break;
+      case 'add:userExample':
+        ruleName = e.data.args[0];
+        var example = e.data.args[1];
+        generator.processExampleFromUser(example, ruleName);
+        break;
+    }
+  }, false);
+
+  function addPiecesToDict(trace, examples) {
+    if (trace.expr.constructor.name === 'Terminal') {
```

Comment:
It's safer to do `trace.expr instanceof ohm.pexprs.Terminal` here. That way, if we ever rename or remove Terminal (unlikely, but it's happened before), it will throw an error.